src/aklite_client_ext.cc

```diff
@@ -120,15 +120,17 @@ GetTargetToInstallResult AkliteClientExt::GetTargetToInstall(const CheckInResult
                << " Skipping its installation.";
     }
 
-    if (force_apps_sync || !client_->appsInSync(Target::fromTufTarget(current))) {
+    auto apps_to_update = client_->appsToUpdate(Target::fromTufTarget(current));
+    if (force_apps_sync || !apps_to_update.empty()) {
       // Force installation of apps
       res.selected_target = current;
       LOG_INFO
           << "The specified Target is already installed, enforcing installation to make sure it's synced and running:"
           << res.selected_target.Name();
 
       res.status = GetTargetToInstallResult::Status::UpdateSyncApps;
-      res.reason = "Syncing Active Target Apps";
+      res.reason =
+          "Syncing Active Target Apps\nApps requiring update:\n" + boost::algorithm::join(apps_to_update, "\n");
```

Comment:
IMO, "Apps requiring update" is redundant after this "Syncing Active Target Apps".
So, I would compose slightly different log:
Instead of
```
 Details:
 | Syncing Active Target Apps
 | Apps requiring update:
 | ml-app (not fetched / not running)
```
produce
```
 Details:
 | Syncing Active Target Apps
 | - ml-app: not fetched / not running
 | - app01: <reason>
 | ...
```

Reply:
Updated.

Reply:
Let's combine commits into a single one.